lib/pco_api/actions.ex

```diff
@@ -18,18 +21,45 @@ defmodule PcoApi.Actions do
 
       def get, do: get("")
       def get(id) when is_integer(id), do: get(Integer.to_string(id))
-      def get(url) when is_binary(url), do: request(:get, url, []) |> new
+      def get(url) when is_binary(url), do: request(:get, url, []) |> to_record
       def get(params) when is_list(params), do: get(params, "")
-      def get(params, url) when is_list(params), do: request(:get, url, params) |> new
+      def get(params, url) when is_list(params), do: request(:get, url, params) |> to_record
 
-      def new(results) when is_list(results), do: results |> Enum.map(&(&1 |> new))
-      def new(%{"id" => id, "links" => links, "attributes" => attrs, "type" => type}) do
+      def create(%PcoApi.Record{attributes: _, type: _} = record), do: create(record, "")
+      def create(%PcoApi.Record{attributes: _, type: _} = record, url) when is_binary(url) do
+        # TODO: Error handling for when the record isn't created
+        record |> PcoApi.Record.to_json |> create(url)
+      end
+      def create(json, url) when is_binary(json) do
```

Comment:
Not a huge fan of the cond nested in a case here.

Could you create a secondary function that pattern matches both out?

I.e. something like:

```def create(json, url) when is_binary(json) do
  url
  |> post(json, [], hackneyL [basic_auth: {PcoApi.key, PcoApi.secret}])
  |> do_create
end

def do_create({:ok, %HTTPoison.Response{status_code: code, body: %{"data" => data}}}), do: data |> to_record
def do_create({:ok, %HTTPoison.Response{status_code: code, body: body}}), do: body
def do_create({:ok, %HTTPoison.Response{body: body}}), do: raise "PcoApi ok, but not ok: #{IO.inspect body}"
def do_create({:error, err}), do: raise "PcoApi error: #{IO.inspect err}"```